Browser_IDE/notifications.js

```diff
@@ -0,0 +1,79 @@
+"use strict";
+
+const NotificationIcons = Object.freeze({
+  CONSTRUCTION:   { class: "bi-box-seam-fill", color: "#FFF" },
+  ERROR:  { class: "bi-exclamation-octagon", color: "#F00" },
+  WARNING: { class: "bi-exclamation-triangle", color: "#FA0" },
+  SUCCESS: { class: "bi-patch-check", color: "#0F0" },
+  INFO: { class: "bi-exclamation-triangle", color: "#FFF" },
+  NONE: { class: "", color: "#FFF" }
+});
+
+// setup area for notifications to appear in
+let notificationsArea = elem("div", { class: "sk-notification-area" });
+document.body.appendChild(notificationsArea);
+
+
+function displayEditorNotification(message, icon=NotificationIcons.NONE, timeout=5, callback=null){
+    // construct notification
+    let notificationIcon = elem("span", { class: icon.class, style: {color: icon.color} });
+
+    let notificationText = elem("div", { class: "sk-notification-body" }, [elemFromText(message)]);
+
+    let notificationCloseButton = elem("button", { class: "bi bi-x-lg" });
+    let notificationCloseButtonWrapper = elem("div", {}, [notificationCloseButton]);
+
+    let notification = elem("div", {
+        class: "sk-contents sk-notification",
+        style: {
+            cursor: callback != null ? "pointer" : ""
+        }
+    }, [
+        notificationIcon,
+        notificationText,
+        notificationCloseButtonWrapper
+    ]);
+
+    let timeoutID = null;
+
+    // setup functions for interaction
+    let deleteNotification = function (){
+        notification.style.pointerEvents = "none";
+        // fadeout to _almost_ fully transparent,
+        // to avoid frames where it's invisible
+        // yet still taking space
+        notification.style.opacity = 0.02;
+
+        clearTimeout(timeoutID);
+
+        // once the fadeout ends, remove the element
+        notification.addEventListener('transitionend', function(event){
+            if (event.propertyName == 'opacity')
+                notificationsArea.removeChild(notification);
+        });
+    }
+
+    let timeoutFunc = function() {
+        deleteNotification();
+    }
+
+    // add events
+    notificationCloseButton.addEventListener("click", function(event) {
+        event.stopPropagation();
+        deleteNotification();
+    });
+
+    notification.addEventListener("click", function(){
+        event.stopPropagation();
+        deleteNotification();
+        if (callback != null)
+            callback();
+    });
```

Comment:
I hadn't noticed that this event handler doesn't actually define the event parameter which is used in its body. Is that legal in JS?

Reply:
Yeah this is one of JavaScript's quirks - extra parameters in a function call just get ignored (like in this case), and missing parameters become `undefined`, so you can really call any function with any number of parameters 😅 Just in case it's interesting, this is actually used to handle the 'overload emulation' in the JavaScript runtime - when a SplashKit function with overloads gets called, the number of arguments passed in gets checked, along with their types, to dispatch the correct C++ overload.

Reply:
The browser's JS runtime will call the handler callback with the event value as argument 0, that I can understand, but how is that argument bound specifically to the identifier 'event' in the function body?

Reply:
Oh sorry I see what you're saying! I forgot I used `event` at all. Honestly that was a mistake on my part - I really should have included it as a parameter.
It works because when event listeners are called back, a global property called `event` is also set (along with being passed in) so I was inadvertently using that - this is pretty bad practice though. 
I'd like to fix it, but another PR just for one line seems a bit much - do you think it's alright to bundle in the change with another unrelated PR? Otherwise I'll submit a PR for it.

Reply:
Ah, I see.

If something prevents you from simply pushing it to the repository without a PR, I would leave it or include it in a more general 'tidying-up' PR.

Reply:
I generally try to avoid pushing directly since it's outside the typical procedure in this unit, but considering it's so small I think it'd be fine, will do!